app/models/content_page.rb

```diff
@@ -1,22 +1,13 @@
-class ContentPage < ApplicationRecord
+class ContentPage < Article 
   acts_as_tree
```

Comment:
I don't think a ContentPage is an Article

A clue is that you still need an attribute called category with the value 'article' to tell the two apart !